### PR TITLE
Show total number of events in psort

### PR DESCRIPTION
**Pull request by joachimmetz**
Show total number of events in psort

### CODE REVIEW

File: plaso/multi_processing/psort.py [hunk 1]
```diff
@@ -742,7 +746,7 @@ def _UpdateProcessingStatus(self, pid, process_status, used_memory):
 
     process = self._processes_per_pid[pid]
 
-    processing_status = process_status.get('processing_status', None)
+    status_indicator = process_status.get('processing_status', None)
```
Comment (Onager) [inline plaso/multi_processing/psort.py]: I think the renaming here is good, given the large number of "status" things (ProcessStatus vs ProcessingStatus), but maybe the definitions need to be renamed to, to STATUS_INDICATOR_IDLE etc. 
These definitions are sometimes called "status" and sometimes called "status_indicator" which is a bit confusing.

File: plaso/cli/status_view.py [hunk 2]
```diff
@@ -307,6 +308,25 @@ def _PrintExtractionStatusUpdateWindow(self, processing_status):
       # We need to explicitly flush stdout to prevent partial status updates.
       sys.stdout.flush()
 
+  def _PrintEventsStatus(self, processing_status):
```
Comment (Onager) [inline plaso/cli/status_view.py]: Just pass the events_status, rather than processing_status as it's not used.

File: plaso/cli/psort_tool.py [hunk 3]
```diff
@@ -508,6 +508,10 @@ def ProcessStorage(self):
           self._storage_file_path))
       return
 
+    total_number_of_events = 0
+    for session in storage_reader.GetSessions():
```
[the remark this answers is not in the document]
Comment (Onager) [reply plaso/cli/psort_tool.py]: Having read more, I think this functionality actually should move to ExportEvents in the psort engine. 
We already pass in the storage reader, and we don't use the total number of events in the tool other than to pass to there. It also means we don't need to change the ExportEvents signature.

File: plaso/engine/processing_status.py [hunk 4]
```diff
@@ -508,6 +518,19 @@ def UpdateWorkerStatus(
         number_of_consumed_reports, number_of_produced_reports)
 
 
+class EventsStatus(object):
+  """The status of the events.
```
Comment (Onager) [inline plaso/engine/processing_status.py]: Please confirm you have plans to add more attributes to this class. Otherwise, we can just use an int and not add another layer of indirection.

File: plaso/engine/processing_status.py [hunk 5]
```diff
@@ -508,6 +518,19 @@ def UpdateWorkerStatus(
...
+
+  Attributes:
+    total_number_of_events (int): total number of events.
```
Comment (Onager) [inline plaso/engine/processing_status.py]: Is this intended to be the total number of events in a store, or just a number of events in general? If the latter, drop the "total_", if something else, please update the docstring to clarify what this is a total of.

File: plaso/engine/processing_status.py [hunk 6]
```diff
@@ -508,6 +518,19 @@ def UpdateWorkerStatus(
...
+  """
+
+  def __init__(self):
+    """Initializes a tasks status."""
```
Comment (Onager) [inline plaso/engine/processing_status.py]: -tasks +events

File: plaso/cli/psort_tool.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (joachimmetz) [reply plaso/cli/psort_tool.py]: ack

File: plaso/cli/psteal_tool.py [hunk 7]
```diff
@@ -191,6 +191,10 @@ def AnalyzeEvents(self):
           self._storage_file_path))
       return
 
+    total_number_of_events = 0
+    for session in storage_reader.GetSessions():
```
[the remark this answers is not in the document]
Comment (joachimmetz) [reply plaso/cli/psteal_tool.py]: ack

File: plaso/cli/status_view.py (hunk 2 above)
Comment (joachimmetz) [reply plaso/cli/status_view.py]: ack

File: plaso/engine/processing_status.py (hunk 4 above)
Comment (joachimmetz) [reply plaso/engine/processing_status.py]: Yes I think we could add other information such as:

* Events MACB grouped
* Duplicate events removed
* Events filtered

File: plaso/cli/status_view.py [hunk 8]
```diff
@@ -307,6 +308,25 @@ def _PrintExtractionStatusUpdateWindow(self, processing_status):
...
+    """Prints the status of the events.
```
[the remark this answers is not in the document]
Comment (joachimmetz) [reply plaso/cli/status_view.py]: I opt to keep as-is to extend EventStatus as indicated below

File: plaso/engine/processing_status.py (hunk 6 above)
Comment (joachimmetz) [reply plaso/engine/processing_status.py]: ack

File: plaso/multi_processing/psort.py (hunk 1 above)
Comment (joachimmetz) [reply plaso/multi_processing/psort.py]: Moved AI to separate issue https://github.com/log2timeline/plaso/issues/2437

File: plaso/engine/processing_status.py (hunk 5 above)
Comment (Onager) [reply plaso/engine/processing_status.py]: Ping for @joachimmetz - I think you missed this comment

File: plaso/multi_processing/psort.py [hunk 9]
```diff
@@ -975,6 +979,7 @@ def AnalyzeEvents(
     if keyboard_interrupt:
       raise KeyboardInterrupt
 
+  # pylint: disable=too-many-arguments
```
Comment (Onager) [inline plaso/multi_processing/psort.py]: Why is this necessary? There's no change to this method, was this not being caught by the linter before for some reason? If not, please remove.

File: plaso/engine/processing_status.py (hunk 5 above)
Comment (Onager) [reply plaso/engine/processing_status.py]: Extra comment to add to code review

Comment (joachimmetz) [reply plaso/engine/processing_status.py]: Total in storage file.

File: plaso/multi_processing/psort.py (hunk 9 above)
Comment (joachimmetz) [reply plaso/multi_processing/psort.py]: no longer needed.
